rust/arrow/src/record_batch.rs

```diff
@@ -75,6 +75,25 @@ impl RecordBatch {
     /// # }
     /// ```
     pub fn try_new(schema: SchemaRef, columns: Vec<ArrayRef>) -> Result<Self> {
+        let options = RecordBatchOptions::default();
+        Self::validate_new_batch(&schema, columns.as_slice(), &options)?;
+        Ok(RecordBatch { schema, columns })
+    }
+
+    pub fn try_new_with_options(
```

Comment:
@alamb @jorgecarleitao I must have done something wrong on the rebase, I had added doc comments for this function, and added tests :(